### ISSUE TITLE
Feature Request: Providing Gradients for `layer.set_weights()`

### ISSUE THREAD

**Issue by NuoWenLei** [tensorflow/tensorflow#53541]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.7.0
- Are you willing to contribute it (Yes/No): No, I am not experienced enough.

**Describe the feature and the current behavior/state.**

The feature is to make `set_weights()` differentiable. The current behavior is that the `model.set_weights()` function breaks the gradient. Given a model named `learner` and a batch of data `batch`

```
l = tf.constant(0.1)
with tf.GradientTape() as gt:
  gt.watch(l)
  new_weights = [w * l for w in learner.weights]

  learner.set_weights(new_weights)

  y_pred = learner(batch["train"]["X"])
  loss = tf.keras.losses.categorical_crossentropy(batch["train"]["y"], y_pred)
grads = gt.gradient(loss, l)
grads # returns None
```

This example returns None for the gradients. In fact, the gradient tape can't calculate the gradient of any variable related to calculating the `new_weights`.

So I was wondering if it's possible to provide gradients for `set_weights()` or if there are any work-arounds to this problem.

**Will this change the current api? How?**

I don't see how this feature would change the current api. It is more a modification on an existing function.

**Who will benefit with this feature?**

I believe adding this feature would make projects related to meta-learning easier to realize.

In my specific case, I was trying to implement [this paper](https://openreview.net/pdf?id=rJY0-Kcll) and translate the [pytorch implementation](https://github.com/markdtw/meta-learning-lstm-pytorch) to tensorflow, but I think this feature would generally benefit many projects that are related to the growing field of meta-learning.

**Any Other info.**
Thanks and Merry Christmas!!!

**Comment by tilakrayal** [comment]
@NuoWenLei ,
Please feel free to close this issue since it is already being tracked in Keras repo? Thanks!
